src/System.IO.Compression/src/System/IO/Compression/DeflateManaged/DeflateManagedStream.cs

```diff
@@ -43,6 +43,17 @@ internal DeflateManagedStream(Stream stream, bool leaveOpen, IFileFormatReader r
             InitializeInflater(stream, leaveOpen, reader);
         }
 
+        // A specific constructor to allow decompression of Deflate64
+        internal DeflateManagedStream(Stream stream, bool deflate64)
```

Comment:
Seems potentially error-prone/confusing to use a bool, as the other ctors use a bool for something completely different, whether to close the stream or not when disposing.  Maybe a CompressionMethodValues?